paxml/programs.py

```diff
@@ -316,6 +325,7 @@ def _sync_device(self, new_state: TrainState, train_step: int):
     # a gap between steps.
     new_train_step = int(
         py_utils.maybe_unreplicate_for_fully_replicated(new_state.step)
+        /self._num_sub_batches
```

Comment:
it's unclear to me how this interacts with 

new_train_step = train_step + 1

on line 293